src/util/getRandomValues.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+
+/*
+ * API shim for `window.crypto.getRandomValues` in the browser, for
+ * compatibility with browser, Node, Jest, and Observable.
+ *
+ * Forked from NPM `get-random-bytes` by Kenan Yildirim, which is
+ * released under the MIT License.
+ */
+
+/**
+ * Fill the given buffer with cryptographically secure random bytes. The
+ * buffer length must not exceed 65536.
+ */
+function getRandomValues(buf: Uint8Array): Uint8Array {
+  if (typeof window !== "undefined") {
+    if (window.crypto && window.crypto.getRandomValues) {
+      return window.crypto.getRandomValues(buf);
+    }
+    if (
+      typeof window.msCrypto === "object" &&
+      typeof window.msCrypto.getRandomValues === "function"
+    ) {
+      return window.msCrypto.getRandomValues(buf);
+    }
+  }
+
+  if (typeof require !== "undefined") {
+    // Late-import `crypto` to avoid `require` in Observable notebooks,
+    // and avoid using a literal `require(...)` to prevent Webpack from
+    // rewriting this. (Getting Webpack externals to work properly is .)
+    //
+    /* eslint-disable camelcase */
+    /* eslint-disable no-undef */
+    const realRequire =
+      // $FlowExpectedError[cannot-resolve-name]
+      typeof __non_webpack_require__ !== "undefined"
```

Comment:
ah this is what we were missing!

Reply:
Yeah; with or without `externals`, Webpack still munges the `require`
logic with its own machinery. I poked at it for a while and opted to
just avoid it. You can work around it without `__non_webpack_require__`
by doing something like `globals["require"](...)`, but that doesn’t work
in Jest because reasons… this block of code is more finicky than it
might seem.